plugin.video.reddit_viewer/resources/lib/utils.py

```diff
@@ -0,0 +1,983 @@
+        if not subreddits_dlist:
+            if os.path.exists(subredditsPickle):
+                subreddits_dlist=load_dict(subredditsPickle)
+
+        subreddit_search=subreddit_entry.lower()
+        if '/' in subreddit_search:
+            subreddit_search=subreddit_search.split('/')[0]
+
+        if '+' in subreddit_search:
+            subreddit_search=random.choice(subreddit_search.split('+'))
+
+        for sd in subreddits_dlist:
+            #we have an entry in our pickle file about the subreddit entry
+            if sd.get('entry_name')==subreddit_search:
+                return sd
+    except:
+        #sometimes we get a race condition when the save thread is saving and the index function is listing
+        #hopefully the 'global' line up above minimizes this
+        pass
+
+def ret_sub_icon(subreddit):
+    sub_info=ret_sub_info(subreddit)
+
+    if sub_info:
+        #return the first item that isn't blank.
+        return next((item for item in [sub_info.get('icon_img'),sub_info.get('banner_img'),sub_info.get('header_img')] if item ), '')
+        #return sub_info.get('icon_img')
+
+subredditsFile_entries=[]
+def load_subredditsFile():
+    global subredditsFile_entries
+    if not subredditsFile_entries:
+        if os.path.exists(subredditsFile):  #....\Kodi\userdata\addon_data\plugin.video.reddit_viewer\subreddits
+            with open(subredditsFile, 'r') as fh:
+                content = fh.read()
+            spl = content.split('\n')
+
+            for i in range(0, len(spl), 1):
+                if spl[i]:
+                    subreddit = spl[i].strip()
+
+                    subredditsFile_entries.append(subreddit )
+    return subredditsFile_entries
+
+def subreddit_in_favorites( subreddit ):
+    sub_favorites=load_subredditsFile()
+    for entry in sub_favorites:
+        if subreddit.lower() == entry.lower():
+            return True
+        if '+' in entry:
+            spl=entry.split('+')
+            for s in spl:
+                if subreddit.lower() == s.lower():
+                    return True
+
+def colored_subreddit(subreddit,color='cadetblue', add_r=True):
+    #return "[COLOR "+color+"]r/" + subreddit + "[/COLOR]"
+    return "[COLOR %s]%s%s[/COLOR]" %(color,('r/' if add_r else ''),subreddit )
+
+def open_web_browser(url,name,type):
+    #http://forum.kodi.tv/showthread.php?tid=235733
+    osWin = xbmc.getCondVisibility('system.platform.windows')
+    osOsx = xbmc.getCondVisibility('system.platform.osx')
+    osLinux = xbmc.getCondVisibility('system.platform.linux')
+    osAndroid = xbmc.getCondVisibility('System.Platform.Android')
+    #url = 'http://www.google.com/'
+
+    import xbmcaddon
+    addon=xbmcaddon.Addon()
+    custom_link_command=addon.getSetting('custom_link_command')
+    if custom_link_command:
+        custom_link_command=custom_link_command.replace('{url}',url)
+        log('Running custom command for link:\n' + custom_link_command)
+        exec( custom_link_command )
```

Comment:
This allows the user to customize which web browser to follow a link posted on reddit.